File: HorseRacing.Web/src/interfaces/api/contracts/model/auth/responses/authentication/authentication-response.ts
import type { BaseResponse } from "~/interfaces/api/contracts/base/responses/base-response";
import type { AuthenticationResponseDto } from "./authentication-response-dto";

export class AuthenticationResponse implements BaseResponse<AuthenticationResponseDto> {
    Data: AuthenticationResponseDto;

    constructor(data?: AuthenticationResponseDto) {
        this.Data = data || { Id: '', UserName: '', Token: '', FirstName: '', LastName: '', Email: '' };
    }
}

File: HorseRacing.Contracts/Models/Authentication/Responses/Authentication/AuthenticationResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Authentication.Responses.Authentication
{
    /// <summary>
    /// Модель ответа при авторизации
    /// </summary>
    public class AuthenticationResponse : BaseResponse<AuthenticationResponseDto>
    {
        public AuthenticationResponse() : base() { }

        public AuthenticationResponse(AuthenticationResponseDto data) : base(data) { }
    }
}
